src/http/ngx_http_wasm_api.c

```diff
@@ -679,18 +765,21 @@ ngx_http_wasm_req_get_headers(ngx_http_request_t *r, int32_t addr, int32_t size_
         ngx_http_wasm_h2_header_t   *wh;
 
         wh = &wasm_h2_header_static_table[i];
+        if (wh->ty == PROXY_WASM_RESPONSE_HEADER) {
```

Comment:
It is weird to check response type in a request header method. Could we just split it into two?

Reply:
I don't think it is necessary to separate the data of request and response for the time being. They use the exact same data structure and can reuse the same query function. I think they have met the current needs by distinguishing them by type. Regarding feeling weird, can we resolve it by judging `PROXY_WASM_REQUEST_HEADER`?

Reply:
I don't think so. It is very strange to mix request & response headers.

Reply:
Why do we need additional checks if we can avoid it? We can use two different variable in the same type.

Reply:
updated